core/include/detray/core/detector.hpp

```diff
@@ -536,9 +545,12 @@ class detector {
 };
 
 /// @brief A static inplementation of detector data for device
-template <typename detector_type>
+template <typename metadata, template <typename> class bfield_t,
+          typename container_t>
 struct detector_view {
 
+    using detector_type = detector<metadata, bfield_t, container_t>;
```

Comment:
Just out of curiosity: Is this needed in the Kalman filter?

Reply:
No. I forgot to remove it

Reply:
Ah but I think we need it for the definition of member variables.
```
    vecmem::data::vector_view<typename detector_type::volume_type>
        _volumes_data;
    vecmem::data::vector_view<typename detector_type::surface_type>
        _surfaces_data;
    typename detector_type::mask_container::view_type _masks_data;
    typename detector_type::material_container::view_type _materials_data;
    typename detector_type::transform_container::view_type _transforms_data;
    typename detector_type::sf_finder_container::view_type _sf_finder_data;
    typename detector_type::volume_finder::view_type _volume_finder_data;
    typename detector_type::bfield_type::view_t _bfield_view;
```

Reply:
Makes sense, thanks